Dashboard crash in 1.9.0

## Bug Report
Experiencing this crash while usual interaction with dashboard.

### Logs
```
panic: runtime error: index out of range [0] with length 0 [recovered]
        panic: runtime error: index out of range [0] with length 0

goroutine 159 [running]:
github.com/rivo/tview.(*Application).Run.func1()
        /.cache/mod/github.com/rivo/tview@v0.0.0-20241103174730-c76f7879f592/application.go:297 +0x45
panic({0x31fa4a0?, 0xc001a64030?})
        /toolchain/go/src/runtime/panic.go:785 +0x132
github.com/siderolabs/talos/internal/pkg/dashboard/components.(*Header).updateNodeAPIData(0xc000a58780, {0xc00000ba80, 0xb}, 0xc00085a180)
        /src/internal/pkg/dashboard/components/header.go:153 +0xbb4
github.com/siderolabs/talos/internal/pkg/dashboard/components.(*Header).OnAPIDataChange(0xc000a58780, {0xc00000ba80, 0xb}, 0xc00188a1e0?)
        /src/internal/pkg/dashboard/components/header.go:86 +0x4c
github.com/siderolabs/talos/internal/pkg/dashboard.(*Dashboard).processAPIData(...)
        /src/internal/pkg/dashboard/dashboard.go:447
github.com/siderolabs/talos/internal/pkg/dashboard.(*Dashboard).startDataHandler.func2.3()
        /src/internal/pkg/dashboard/dashboard.go:406 +0x5f
github.com/siderolabs/talos/internal/pkg/dashboard.(*Dashboard).startDataHandler.func2.(*Application).QueueUpdateDraw.6()
        /.cache/mod/github.com/rivo/tview@v0.0.0-20241103174730-c76f7879f592/application.go:862 +0x22
github.com/rivo/tview.(*Application).Run(0xc000166fc0)
        /.cache/mod/github.com/rivo/tview@v0.0.0-20241103174730-c76f7879f592/application.go:485 +0xd55
github.com/siderolabs/talos/internal/pkg/dashboard.Run.func2()
        /src/internal/pkg/dashboard/dashboard.go:341 +0x3e
golang.org/x/sync/errgroup.(*Group).Go.func1()
        /.cache/mod/golang.org/x/sync@v0.10.0/errgroup/errgroup.go:78 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 1
        /.cache/mod/golang.org/x/sync@v0.10.0/errgroup/errgroup.go:75 +0x96
```

### Environment

- Talos version: 1.9.0 (some nodes were 1.8.3)
- Kubernetes version: 1.30.7
- Platform: Bare Metal

--- a/internal/pkg/dashboard/components/header.go
+++ b/internal/pkg/dashboard/components/header.go
@@ -149,8 +149,6 @@ func (widget *Header) updateNodeAPIData(node string, data *apidata.Node) {
 
 		if numCPUs > 0 {
 			nodeData.cpuFreq = fmt.Sprintf("%dx%s", numCPUs, widget.humanizeCPUFrequency(data.CPUsInfo.GetCpuInfo()[0].GetCpuMhz()))
-		} else {
-			nodeData.cpuFreq = widget.humanizeCPUFrequency(data.CPUsInfo.GetCpuInfo()[0].GetCpuMhz())
 		}
 	}
 

